prefect_gcp/cloud_storage.py

```diff
@@ -452,14 +451,195 @@ def example_cloud_storage_copy_blob_flow():
         dest_blob = source_blob
 
     source_blob_obj = source_bucket_obj.blob(source_blob)
-    partial_copy_blob = partial(
+    await run_sync_in_worker_thread(
         source_bucket_obj.copy_blob,
         blob=source_blob_obj,
         destination_bucket=dest_bucket_obj,
         new_name=dest_blob,
         timeout=timeout,
         **copy_kwargs,
     )
-    await to_thread.run_sync(partial_copy_blob)
 
     return dest_blob
+
+
+class GcsBucket(ReadableFileSystem, WritableFileSystem):
+    """
+    Block used to store data using GCP Cloud Storage Buckets.
+
+    Attributes:
+        bucket: Name of the bucket.
+        gcp_credentials: The credentials to authenticate with GCP.
+        basepath: Used when you don't want to read/write at base level.
+
+    Example:
+        Load stored GCP Cloud Storage Bucket:
+        ```python
+        from prefect_gcp import GcpCloudStorageBucket
+        gcp_cloud_storage_bucket_block = GcpCloudStorageBucket.load("BLOCK_NAME")
+        ```
+    """
+
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/4CD4wwbiIKPkZDt4U3TEuW/c112fe85653da054b6d5334ef662bec4/gcp.png?h=250"  # noqa
+    _block_type_name = "GCS Bucket"
+
+    bucket: str
+    gcp_credentials: GcpCredentials
+    basepath: str = ""
```

Comment:
We need some descriptions for these fields. Could be a good opportunity to pull in the block standard test suite.